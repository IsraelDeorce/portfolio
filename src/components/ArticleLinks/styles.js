import styled from 'styled-components'

export const ArticleLinks = styled.ul`
  display: flex;
  flex-direction: column;
  padding-left: 1em;
  & > * {
    color: ${({ theme }) => theme.c_brackets};
    &:hover {
      color: ${({ theme }) => theme.c_varType};
    }
    padding-bottom: 1em;
  }
`
