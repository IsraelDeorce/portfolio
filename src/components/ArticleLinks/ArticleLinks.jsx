import { useTranslation } from 'react-i18next'
import { Link } from 'react-router-dom'
import * as S from './styles'

function ArticleLinks() {
  const { t } = useTranslation()

  return (
    <S.ArticleLinks>
      <Link to='/blog/time-complexity'>{t('articles.time_complexity')}</Link>
      <Link to='/blog/software-engineering'>{t('skills.software_engineering')}</Link>
      <Link to='/blog/data-structures'>{t('articles.data_structures')}</Link>
      <Link to='/blog/algorithms'>{t('articles.algorithms')}</Link>
    </S.ArticleLinks>
  )
}

export default ArticleLinks
