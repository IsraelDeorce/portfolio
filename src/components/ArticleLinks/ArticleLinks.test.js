import ArticleLinks from './ArticleLinks'

describe('<ArticleLinks/>', () => {
  afterEach(() => jest.clearAllMocks())

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<ArticleLinks />)).toMatchSnapshot()
    })
  })
})