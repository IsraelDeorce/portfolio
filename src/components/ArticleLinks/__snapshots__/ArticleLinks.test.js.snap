// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleLinks/> Snapshot renders without crashing using shallow 1`] = `
<styled.ul>
  <Link
    to="/blog/big-o"
  >
    articles.big_o
  </Link>
  <Link
    to="/blog/data-structures"
  >
    skills.software_engineering
  </Link>
  <Link
    to="/blog/data-structures"
  >
    articles.data_structures
  </Link>
  <Link
    to="/blog/data-structures"
  >
    articles.algorithms
  </Link>
</styled.ul>
`;
